--- a/peft/peft_model.py
+++ b/peft/peft_model.py
@@ -717,7 +717,7 @@
 
     def forward(
         self,
-        input_ids=None,
+        input_values=None,
         attention_mask=None,
         inputs_embeds=None,
         labels=None,
@@ -730,9 +730,8 @@
         peft_config = self.active_peft_config
         if not peft_config.is_prompt_learning:
             return self.base_model(
-                input_ids=input_ids,
+                input_values=input_values,
                 attention_mask=attention_mask,
-                inputs_embeds=inputs_embeds,
                 labels=labels,
                 output_attentions=output_attentions,
                 output_hidden_states=output_hidden_states,
@@ -740,7 +739,7 @@
                 **kwargs,
             )
 
-        batch_size = _get_batch_size(input_ids, inputs_embeds)
+        batch_size = _get_batch_size(input_values, inputs_embeds)
         if attention_mask is not None:
             # concat prompt attention mask
             prefix_attention_mask = torch.ones(batch_size, peft_config.num_virtual_tokens).to(attention_mask.device)
@@ -759,7 +758,7 @@
         )
 
         if peft_config.peft_type == PeftType.PREFIX_TUNING:
-            return self._prefix_tuning_forward(input_ids=input_ids, **kwargs)
+            return self._prefix_tuning_forward(input_values=input_values, **kwargs)
         else:
             if kwargs.get("token_type_ids", None) is not None:
                 kwargs["token_type_ids"] = torch.cat(
@@ -770,7 +769,7 @@
                     dim=1,
                 ).long()
             if inputs_embeds is None:
-                inputs_embeds = self.word_embeddings(input_ids)
+                inputs_embeds = self.word_embeddings(input_values)
             prompts = self.get_prompt(batch_size=batch_size)
             prompts = prompts.to(inputs_embeds.dtype)
             inputs_embeds = torch.cat((prompts, inputs_embeds), dim=1)
@@ -778,7 +777,7 @@
 
     def _prefix_tuning_forward(
         self,
-        input_ids=None,
+        input_values=None,
         attention_mask=None,
         inputs_embeds=None,
         labels=None,
@@ -787,14 +786,13 @@
         return_dict=None,
         **kwargs,
     ):
-        batch_size = _get_batch_size(input_ids, inputs_embeds)
+        batch_size = _get_batch_size(input_values, inputs_embeds)
         past_key_values = self.get_prompt(batch_size)
         fwd_params = list(inspect.signature(self.base_model.forward).parameters.keys())
         kwargs.update(
             {
-                "input_ids": input_ids,
+                "input_values": input_values,
                 "attention_mask": attention_mask,
-                "inputs_embeds": inputs_embeds,
                 "output_attentions": output_attentions,
                 "output_hidden_states": output_hidden_states,
                 "return_dict": return_dict,
@@ -891,7 +889,7 @@
 
     def forward(
         self,
-        input_ids=None,
+        input_values=None,
         attention_mask=None,
         inputs_embeds=None,
         labels=None,
@@ -906,9 +904,8 @@
                 if inputs_embeds is not None:
                     raise AssertionError("forward in MPTForCausalLM does not support inputs_embeds")
                 return self.base_model(
-                    input_ids=input_ids,
+                    input_values=input_values,
                     attention_mask=attention_mask,
-                    labels=labels,
                     output_attentions=output_attentions,
                     output_hidden_states=output_hidden_states,
                     return_dict=return_dict,
@@ -916,17 +913,15 @@
                 )
 
             return self.base_model(
-                input_ids=input_ids,
+                input_values=input_values,
                 attention_mask=attention_mask,
-                inputs_embeds=inputs_embeds,
-                labels=labels,
                 output_attentions=output_attentions,
                 output_hidden_states=output_hidden_states,
                 return_dict=return_dict,
                 **kwargs,
             )
 
-        batch_size = _get_batch_size(input_ids, inputs_embeds)
+        batch_size = _get_batch_size(input_values, inputs_embeds)
         if attention_mask is not None:
             # concat prompt attention mask
             prefix_attention_mask = torch.ones(batch_size, peft_config.num_virtual_tokens).to(attention_mask.device)
@@ -951,11 +946,11 @@
         if peft_config.peft_type == PeftType.PREFIX_TUNING:
             past_key_values = self.get_prompt(batch_size)
             return self.base_model(
-                input_ids=input_ids, inputs_embeds=inputs_embeds, past_key_values=past_key_values, **kwargs
+                input_values=input_values, past_key_values=past_key_values, **kwargs
             )
         else:
             if inputs_embeds is None:
-                inputs_embeds = self.word_embeddings(input_ids)
+                inputs_embeds = self.word_embeddings(input_values)
             # concat prompt labels
             if labels is not None:
                 prefix_labels = torch.full((batch_size, peft_config.num_virtual_tokens), -100).to(labels.device)
@@ -986,8 +981,8 @@
         if peft_config.is_prompt_learning:
             if model_kwargs.get("attention_mask", None) is not None:
                 prefix_attention_mask = torch.ones(
-                    model_kwargs["input_ids"].shape[0], peft_config.num_virtual_tokens
-                ).to(model_kwargs["input_ids"].device)
+                    model_kwargs["input_values"].shape[0], peft_config.num_virtual_tokens
+                ).to(model_kwargs["input_values"].device)
                 model_kwargs["attention_mask"] = torch.cat(
                     (prefix_attention_mask, model_kwargs["attention_mask"]), dim=1
                 )
@@ -1003,15 +998,15 @@
                 kwargs["token_type_ids"] = None
 
             if model_kwargs["past_key_values"] is None and peft_config.peft_type == PeftType.PREFIX_TUNING:
-                past_key_values = self.get_prompt(batch_size=model_kwargs["input_ids"].shape[0])
+                past_key_values = self.get_prompt(batch_size=model_kwargs["input_values"].shape[0])
                 model_kwargs["past_key_values"] = past_key_values
             else:
                 if model_kwargs["past_key_values"] is None:
-                    inputs_embeds = self.word_embeddings(model_kwargs["input_ids"])
-                    prompts = self.get_prompt(batch_size=model_kwargs["input_ids"].shape[0])
+                    inputs_embeds = self.word_embeddings(model_kwargs["input_values"])
+                    prompts = self.get_prompt(batch_size=model_kwargs["input_values"].shape[0])
                     prompts = prompts.to(inputs_embeds.dtype)
                     model_kwargs["inputs_embeds"] = torch.cat((prompts, inputs_embeds), dim=1)
-                    model_kwargs["input_ids"] = None
+                    model_kwargs["input_values"] = None
 
         return model_kwargs
 
